dashboard/app/controllers/api/v1/users_controller.rb

```diff
@@ -189,6 +190,22 @@ def update_display_theme
     render json: {display_theme: @user.display_theme}
   end
 
+  # POST /api/v1/users/<user_id>/ai_tutor_access
+  def update_ai_tutor_access
+    return head :unauthorized unless current_user&.teacher?
+    target_user = User.find_by_id(params[:user_id])
+
+    unless target_user && target_user.student_of?(current_user)
+      return head :unauthorized
+    end
+
+    target_user.ai_tutor_access_denied = !params[:ai_tutor_access]
+    target_user.save
+
+    # TODO: Do we need to handle errors here?
```

Comment:
I was thinking this would look like
```
if target_user.save
  ...
else
  ...
```
But none of the other methods in this file do this.